subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/downstream/test/TestDownstreamPlugin.java

```diff
@@ -22,12 +35,36 @@
 
     public void apply(final Project project) {
         IncubatingWarning.warn("downstream-testing plugin");
-        TaskMaker.task(project, TEST_DOWNSTREAM_TASK, TestDownstreamTask.class, new Action<TestDownstreamTask>() {
+
+        project.getPlugins().apply(UploadGistsPlugin.class);
+
+        TestDownstreamTask testDownstreamTask = TaskMaker.task(project, TEST_DOWNSTREAM_TASK, TestDownstreamTask.class, new Action<TestDownstreamTask>() {
             @Override
             public void execute(TestDownstreamTask task) {
                 task.setDescription("Runs all downstream tests.");
             }
         });
+
+        final UploadGistsTask uploadGistsTask = (UploadGistsTask) project.getTasks().findByName(UploadGistsPlugin.UPLOAD_GISTS_TASK);
+
+        uploadGistsTask.setFilesToUpload((ConfigurableFileTree)
+            project
+                .fileTree(testDownstreamTask.getLogDirectory())
+                .include("*.log")
+        );
+
+        // it has to be deferred because TestDownstreamReleaseTask are added by user after applying this plugin
+        DeferredConfiguration.deferredConfiguration(project, new Runnable() {
```

Comment:
A couple of options:
1. Add this to TravisPlugin. E.g. travis plugin would configure 'finalizedBy' only when the build is invoked on Travis CI. If the build is invoked locally, we don't send gists
2. TravisPlugin checks if we are on CI machine and if so, we are making uploadGistsTask.enabled = false
3. Similar to previous point but flipping the default. By default uploadGists is disabled. Only TravisPlugin enables it when we're running on Travis.

I think I like option 3) most. There are more options of course :) please suggest!